torch/CMakeLists.txt

```diff
@@ -288,47 +188,60 @@ target_compile_options(torch
   -Wno-missing-braces # This warning is buggy
   -Wno-unknown-pragmas)
 
-if ($ENV{WERROR})
+if ($ENV{TORCH_WERROR})
   target_compile_options(torch PRIVATE -Werror)
+else()
+  target_compile_options(torch PRIVATE -Wno-error)
 endif()
 
 target_link_libraries(torch
-  ${TORCH_CUDA_LIBRARIES}
-  ${CAFFE2_LIBRARY}
-  ${PROTOBUF_LIBRARY}
-  protobuf-nanopb
-)
+  caffe2_library
+  protobuf-nanopb)
 if(USE_CUDA)
-  if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
-    target_link_libraries(torch -Wl,--no-as-needed ${CAFFE2_GPU_LIBRARY} -Wl,--as-needed)
-  else()
-    target_link_libraries(torch ${CAFFE2_GPU_LIBRARY})
-  endif()
+  target_link_libraries(torch caffe2_gpu_library)
+endif()
+if(USE_ROCM)
+  target_link_libraries(torch caffe2_hip_library)
+endif()
+
+set(TH_CPU_INCLUDE
+  # dense
+  ${TORCH_SRC_DIR}/../aten/src/TH
+  ${CMAKE_CURRENT_BINARY_DIR}/../aten/src/TH
+  ${TORCH_SRC_DIR}/../aten/src
+  ${CMAKE_CURRENT_BINARY_DIR}/../aten/src
+  ${CMAKE_BINARY_DIR}/aten/src)
+target_include_directories(torch PRIVATE ${TH_CPU_INCLUDE})
```

Comment:
If you're using the official `caffe2_gpu_library` target, aren't these include paths supposed to be automatically added?

Reply:
very possibly. I'll try it out

Reply:
tried it, didn't work. it may be an opportunity for a future refactor